Setup guide on homepage

dennisreimann opened the pull request:
The basic two steps are displayed, guiding the user to setup a store and afterwards a wallet.

![empty](https://user-images.githubusercontent.com/886/148270526-2cf14e9a-a391-4cc5-a919-ac1e99e3c116.png)

I would not add more steps, because it gets tricky real fast: The current logic checks and sees if there is at least one store with an onchain wallet or Lightning node configured — if that's the case it is assumed the user knows how things are done and we can hide the guide.

TODO: Figure out what to display once the setup steps are done. Ideas for a MVP-dashboard welcome! /cc @dstrukt

dstrukt commented:
Agreed w.r.t complexity. 

Was trying to map out a more complex setup guide in Figma with the thinking off to the side, and I can almost imagine hearing you sigh when you saw that exploration .. maybe eventually we can help guide users more in the future for specific applications ... but for this release, love the simplicity!

However, w.r.t "Create Store" for no stores. I'm still not completely sure if I'm sold on the in-between state vs. a full page view post sign-up.

<img width="2499" alt="Screen Shot 2022-01-05 at 5 10 08 PM" src="https://user-images.githubusercontent.com/6250771/148312121-69959c7d-cbb5-4796-aa45-4a164d62303a.png">

Despite which direction we go, I do love the change to "Create Store" if no store, was going to mention we should address that (as the previous dropdown was odd) if possible, and you're way ahead of me.

If we do go with this direction, small nit, might be worth making the "Create Store" a primary button to draw more attention to the action.

Regardless, great work!

dennisreimann commented:
Updated the button.

![image](https://user-images.githubusercontent.com/886/148403638-083e60c9-d783-4c88-b2f7-57ee6a604b06.png)

Added a recent notifications "widget" to the homepage, for the case that everything is set up and the guide does not get displayed.

![image](https://user-images.githubusercontent.com/886/148403862-ade52ef6-f81c-4425-9333-fd71020984dd.png)

dstrukt commented:
Discussed in design call.

Plan to move forward with the latest proposed changes.

ACK!

dennisreimann commented:
Progressed as discussed in yesterdays design meeting. /cc @dstrukt @Zaxounette 

There is now also a new store home page, where one lands after creating a store. It contains a setup guide similar to the general homepage, but focused on the store. This is supposed to guide the user towards setting up at least one kind of wallet - onchain or Lightning.

![store-home-1](https://user-images.githubusercontent.com/886/148586886-e35507ac-1dd9-40ab-a289-3d720d36e86c.png)

Once the setup is done, the "What's Next" part appears. It contains the basic information about the core features and how they can be used. 

![store-home-2](https://user-images.githubusercontent.com/886/148586895-098e2a59-e370-444e-968b-a8d650c293f2.png)

Feedback welcome!

dstrukt commented:
Yes!

1. A question that didn't, and maybe should have, occurred to me in the call .. should the Lightning Node (internal) be enabled by default? To potentially skip the empty state page when clicking on the sidebar tab, and remove the extra step here? Maybe there are details here I'm missing.

Wasn't 100% sure what we decided on that one, but something we could consider as well...?

2. The additional copy is interesting, I like selections you chose, and I think with a little bit of finessing this could be a really solid introduction to the power of btcpay within the app, as we discussed ... will follow up on that section with some simple formatting / think on what you're proposing a bit more.

dennisreimann commented:
To point 1: We can only do this under certain circumstances – by default the Internal Node is only available to the instance admin. 

I guess this is also the reason why we haven't set it up automatically yet.

pavlenex commented:
This is excellent work. I'm sure in the future we may consider ironing it out based on feedback, but as minimum viable product, this is imo doing what it's supposed to do.

I am a bit unsure about the `what's next section` mostly because it's text heavy and we still haven't really cleaned things up when it comes to apps, plugins, etc, so introducing all these concept in this way, to me, is a bit too much. First of all, text heavy means more things for translators to translate, and even then these will most likely change in the future.

So I propose that what's next, at this stage of us coming up with an MVP is a bit out of scope. I believe the proper time to add this would be after the cleanup (apps in plugins, etc)

dennisreimann commented:
@pavlenex I agree that it seems too text heavy, but also thought it might be better than nothing -- however, we can remove it for the first iteration and wait for what the dashboard brings.

As for the setup guides Home vs. Store Home: It's a bit tricky, because on the general homepage we have to deal with the possibility of multiple stores. So the check there is more like "is there a store? If so, is there at least one store with a wallet or lightning node?" (assuming the user know the setup process in case he has configured one at least once).
On the Store Home we can focus on that particular store, hence also guide the user towards setting up both, onchain and Lightning. That's my reasoning behind that difference.

pavlenex commented:
I think the problematic thing there is possibly naming `Home`, I expect to go whatever home is when I hit the logo, and was very annoyed that `home` in store kept leading me to something that didn't feel like home. So possibly `dashboard` vs `launchpad` naming would have to be done there to differentiate what we mean, otherwise the flow might be a bit odd to people. I'm not sure if this has been discussed on the  design meeting and decided on, so I'm just raising my two cents that it may not be what people are expecting to get.

When I hit home, I'd expect to land on a dashboard, but perhaps store check should be `launchpad` or similar, as I once[ pointed out on the UX chat](https://chat.btcpayserver.org/btcpayserver/pl/qn3eiskmg7fiumiubn4ee9pk9r) , but I am not even entirely sold this is clear, so def requires wider discussion.

dennisreimann commented:
Agree and think we can call it dashboard once we have something meaningful for the state when the store is set up.

dstrukt commented:
The original intention was to provide a "Home" for each store, where you will eventually see all relevant metrics (for that day, week, etc..) and other actions that will evolve over time, to start orienting the user to this tab.

Clicking on the logo will take you to the "Home" tab of the store instance you're currently in, as well as clicking on "Home" on the sidebar, intended functionality.

However, I could also see "Dashboard" being used as well, and prefer that to "Launchpad" personally.

pavlenex commented:
Just noticed that I get 403 when trying to set up a Lightning Node before the actual on-chain wallet.

NicolasDorier commented:
Can you also rebase, the integration tests error is on CanSendIPN which was flaky because of a bug I fixed.

pavlenex commented on BTCPayServer/Views/Home/Home.cshtml:
```diff
@@ -1,84 +1,104 @@
-                <h3>Easy</h3>
-                <p class="text-muted mb-0">A user-friendly Bitcoin checkout page for your customers.</p>
+        <div class="col-lg-9 col-xl-6">
+            <p class="lead text-secondary">To start accepting transactions, you need to set up a store and wallet.</p>
```
```suggestion
            <p class="lead text-secondary">To start accepting payments, set up a store and a wallet.</p>
```

pavlenex commented on BTCPayServer/Views/Stores/Dashboard.cshtml:
```diff
@@ -0,0 +1,118 @@
+}
+else
+{
+    <p class="lead text-secondary">To start accepting transactions, you need to set up a wallet or Lightning node.</p>
```
```suggestion
    <p class="lead text-secondary">To start accepting payments, set up a wallet or a Lightning node.</p>
```

pavlenex commented:
Just quickly tested things out again, good improvements 🎉 

This flow to me still feels very weird. I don't really understand why the setups are different between `dashboard` and `home button`. Perhaps I am still misunderstanding the big idea behind it.

![2022-01-12_14-03-35](https://user-images.githubusercontent.com/36959754/149145895-36aed989-3fc6-40ef-8979-6a9117f12cb2.gif)

dennisreimann commented:
@pavlenex I described it [here](https://github.com/btcpayserver/btcpayserver/pull/3255#issuecomment-1008125529). What do you recommend?

pavlenex commented:
@dennisreimann Haven't thought to much about it, but it seems that setup flows should at least try to be consistent.

As you can see in a gif [here](https://github.com/btcpayserver/btcpayserver/pull/3255#issuecomment-1011026274) when I have no stores, and I try to set up my first store, the setup flows seem different. On dashboard I have the ability to set up wallet directly leading me to on-chain screen, or lightning, directly leading me to the LN node. Comparing it to the `store home` view I only see wallet setup which leads me to the payment method page. 

I think the first flow is a bit more intuitive. Is there a reason why these two setup flows are different?

One more thing I noticed, is when I have a store already and want to set up a new one, the `dashboard` setup flow no longer appears and I only see notifications even though i haven't set up the store yet.

![2022-01-12_14-11-12 (1)](https://user-images.githubusercontent.com/36959754/149147313-ab01b928-4f0a-418e-8079-85c6de967096.gif)

NicolasDorier commented:
Is it normal: 

![image](https://user-images.githubusercontent.com/3020646/149268279-f2354902-574d-4eef-86bc-de6e6a6d80c9.png)

NicolasDorier commented:
![image](https://user-images.githubusercontent.com/3020646/149268404-18045c01-7270-457f-970d-2b86c06180b7.png)

NicolasDorier commented:
You think improving in next PR? it is a bit strange, Dashboard doesn't have an icon and done steps don't have a check mark.

![image](https://user-images.githubusercontent.com/3020646/149268956-1df26acb-a0d4-4af7-95bf-7d4bcebc8c1c.png)

Also, I think in maybe later PR, when a step is setup by clicking on the dashboard, when the step is done, it should return back to the dashboard, not to the store's settings.

NicolasDorier commented:
It seems to me the ListStores page should be removed at this point?

NicolasDorier commented on BTCPayServer/Controllers/UserStoresController.cs:
```diff
@@ -82,7 +82,7 @@ public async Task<IActionResult> DeleteStorePost(string storeId)
 
         [HttpGet]
-        [Authorize(AuthenticationSchemes = AuthenticationSchemes.Cookie, Policy = Policies.CanViewStoreSettings)]
+        [Authorize(AuthenticationSchemes = AuthenticationSchemes.Cookie, Policy = Policies.CanModifyStoreSettingsUnscoped)]
```
There shouldn't have any Policy here I think.

dennisreimann replied:
If we omit it, the cookie auth handler does not populate CurrentStore. Shall I remove the policy for the list and create actions and resolve the CurrentStore inside the actions?

dennisreimann commented:
The missing icons are a caching problem, can you do a hard reload or clear the cache?

The disabled Internal Node isn't visible, because we switched the colors of the sidebar and main content – the latter having a plain white background now. Will it stay this way, @dstrukt?

NicolasDorier commented:
@dennisreimann the problem was indeed caching. That's weird though, why would have a caching issue? I think all resources have a `v=` query string to prevent this.

NicolasDorier commented:
Ah ok, not in the svg. I will check if I can't add it automatically

NicolasDorier commented:
Shouldn't have anymore cache issue with https://github.com/btcpayserver/btcpayserver/pull/3298
